src/ui/components/Overlay/index.js

```diff
@@ -1,80 +1,96 @@
+/* @flow */
+import invariant from 'invariant';
 import makeClassName from 'classnames';
 import * as React from 'react';
-import PropTypes from 'prop-types';
+import { compose } from 'redux';
+
+import withUIState from 'core/withUIState';
 
 import './styles.scss';
 
-export default class Overlay extends React.Component {
-  static propTypes = {
-    children: PropTypes.node,
-    className: PropTypes.string,
-    onEscapeOverlay: PropTypes.func,
-    visibleOnLoad: PropTypes.bool,
-  };
+type Props = {|
+  className?: string,
+  children: React.Element<any>,
+  id: string,
+  onEscapeOverlay?: () => void,
+  visibleOnLoad?: boolean,
+|};
+
+type UIStateType = {|
+  visible: boolean,
+|};
 
+const initialUIState: UIStateType = { visible: false };
+
+type InternalProps = {|
+  ...Props,
+  setUIState: ($Shape<UIStateType>) => void,
+  uiState: UIStateType,
+|};
+
+export class OverlayBase extends React.Component<InternalProps> {
   static defaultProps = {
     visibleOnLoad: false,
   };
 
-  constructor(props) {
-    super(props);
-    this.state = { visible: props.visibleOnLoad };
-  }
+  componentWillReceiveProps(nextProps: InternalProps) {
+    const { uiState: uiStateOld } = this.props;
+    const { visibleOnLoad: visibleOnLoadNew } = nextProps;
 
-  componentWillReceiveProps(nextProps) {
-    if (nextProps.visibleOnLoad !== undefined) {
-      this.setState({ visible: nextProps.visibleOnLoad });
+    if (
+      (visibleOnLoadNew && !uiStateOld.visible) ||
+      uiStateOld.visible !== nextProps.uiState.visible
```

Comment:
It seems in order to get this to work, I need to reset the state in componentWillUnMount:

```javascript
 componentWillUnmount() {
    this.props.setUIState({ visible: false });
  }
```

otherwise, I can't always reopen it. Egs:

on the detail page, clicking on star (similar thing happens with profile page):
click 1 - it opens
click 2 - nothing happens
click 3 - it opens

on review page, clicking on "edit my review" link:
click 1 - it opens
click 2, etc - nothing happens

I do see the clicking on background working if i click near the top of the page(seems a little sporadic too)

**** Actually after testing this a bit more this (unmount noted above) only seems to work if you close via submitting. Using the background click does not unmount it so i'm still investigating this :/

Reply:
Regarding your question about about `visibleOnLoad`..these 2 components that are set up to use it are always passing this in as true.  These components arent mounted until their respective state in their components show true. so I don't think it's ever changes value at least in these 2 cases. if that makes sense..